mod_reforged/config/make_weapon_loaded.nut

```diff
@@ -0,0 +1,120 @@
+::Reforged.Items <- {}
+
+// This function adjusts a loaded-weapon with the essential functions and variables to support this loading playstyle. Examples are Crossbow and Handgonne
+::Reforged.Items.makeWeaponLoaded <- function( o )
+{
+	o.m.IsLoaded <- false;	// In vanilla this is true. But we don't want to hand out free ammunition
+	o.m.LoadedAmmunitionID <- null;		// Is used for new ammunition types to correctly apply their effects
+	o.m.LoadedAmmunitionCost <- 1;	// Is used so we can refund the cost after battle without having to find the quiver it was from
+
+	if (!"isLoaded" in o)
+	{
+		o.isLoaded <- function()
+		{
+			return this.m.IsLoaded;
+		}
+	}
+
+	if (!"setLoaded" in o)
+	{
+		o.setLoaded <- function( _loaded, _ammoID = null, _ammoCost = 1 )
```

Comment:
Why do we allow `null` as ammoID? Shouldn't we always require an ammoID?

Reply:
Part of this being defaulted to null is to make the function backwards compatible with what vanilla and vanilla mods may expect.

This ID is currently only meant as an optional tool for new quiver variants to use. They can at any time check whether the loaded ammunition came from their quiver/pouch and only then apply their passive effects.